nomos-services/key-management-system/src/keys/ed25519.rs

```diff
@@ -1,20 +1,27 @@
-use bytes::Bytes;
+use bytes::Bytes as RawBytes;
 use ed25519_dalek::ed25519::signature::Signer as _;
-use overwatch::DynError;
 use serde::{Deserialize, Serialize};
 use zeroize::ZeroizeOnDrop;
 
-use crate::secure_key::SecuredKey;
+use crate::keys::{secured_key::SecuredKey, KeyError};
 
 #[derive(Serialize, Deserialize, ZeroizeOnDrop)]
 pub struct Ed25519Key(pub(crate) ed25519_dalek::SigningKey);
 
 impl SecuredKey for Ed25519Key {
-    fn sign(&self, data: Bytes) -> Result<Bytes, DynError> {
-        Ok(Bytes::copy_from_slice(&self.0.sign(&data).to_bytes()))
+    type EncodingFormat = crate::encodings::Bytes;
+    type Error = KeyError;
+
+    fn sign(&self, data: Self::EncodingFormat) -> Result<Self::EncodingFormat, Self::Error> {
+        let data_bytes = data.as_bytes();
+        let signature_slice = self.0.sign(data_bytes).to_bytes();
+        let signature_bytes = RawBytes::copy_from_slice(&signature_slice);
+        Ok(Self::EncodingFormat::from(signature_bytes))
     }
 
-    fn as_pk(&self) -> Bytes {
-        Bytes::copy_from_slice(self.0.verifying_key().as_bytes())
+    fn as_pk(&self) -> Self::EncodingFormat {
```

Comment:
Should this also return a reference to the public key? The caller could then decide if they want to clone/copy in case they need to store it somewhere, but if they need, for instance, it's byte representation or its size, then having a reference should be enough.

Reply:
Probably not (?) or we would need `Cow`, usually pk is not stored but derived from `sk`.

Reply:
But why return `Self::EncodingFormat` instead of returning `Self::PublicKey`? Or `&Self::PublicKey`?

Reply:
Ah, I understood something else. You are right indeed. Its much clearer to have a pk type. I think this comes from the old bytes for everything old style.